app/components/stats/KillsTable.js

```diff
@@ -71,11 +71,17 @@ export default class KillsTable extends Component {
     const killingPunishesByEndFrame = _.keyBy(killingPunishes, 'endFrame');
     const punishThatEndedStock = killingPunishesByEndFrame[stock.endFrame];
 
-    if (!punishThatEndedStock) {
+    if(!punishThatEndedStock ) {
       return <span className={styles['secondary-text']}>Self Destruct</span>;
     }
+    
 
     const lastMove = _.last(punishThatEndedStock.moves);
+
+    if((_.last(punishThatEndedStock.moves)) == undefined){
```

Comment:
You don't need to rerun the same code that just ran. Additionally it's not necessary to check explicitly for undefined. Also never use `==`, only `===`. But in this case you should just do `if (!lastMove)`